include/aspect/utilities.h

```diff
@@ -69,33 +63,329 @@ namespace aspect
+
+
+
+    /**
+     * AsciiDataLookup reads in files containing input data
+       in ascii format. Note the required format of the
+       input data: The first lines may contain any number of comments
+       if they begin with '#', but one of these lines needs to
+       contain the number of grid points in each dimension as
+       for example '# POINTS: 3 3'. The order of the columns
+       has to be 'coordinates data' with @p dim coordinate columns
+       and @p components data columns. Note that the data in the input
+       files need to be sorted in a specific order:
+       the first coordinate needs to ascend first,
+       followed by the second and so on in order to
+       assign the correct data to the prescribed coordinates.
+     */
+    template <int dim>
+    class AsciiDataLookup
+    {
+      public:
+        AsciiDataLookup(const unsigned int components,
+                        const double scale_factor);
+
+        /**
+         * Loads a data text file. Throws an exception if the file does not exist,
+         * if the data file format is incorrect or if the file grid changes over model runtime.
+         */
+        void
+        load_file(const std::string &filename);
+
+        /**
+         * Returns the computed data (velocity, temperature, etc. - according to the used plugin)
+         * in cartesian coordinates.
+         *
+         * @param position The current position to compute the data (velocity, temperature, etc.)
+         * @param time_weight A weighting between the two current timesteps n and n+1
+         */
+        double
+        get_data(const Point<dim> &position,
+                 const unsigned int component) const;
+
+      private:
+        /**
+         * The number of data components read in (=columns in the data file).
+         */
+        const unsigned int components;
+
+        /**
+         * Interpolation functions to access the data.
+         */
+        std::vector<Functions::InterpolatedUniformGridData<dim> *> data;
+
+        /**
+         * Min and Max coordinates in data file
+         */
+        std_cxx11::array<std::pair<double,double>,dim> grid_extent;
+
+        /**
+         * Number of points in the data grid.
+         */
+        TableIndices<dim> table_points;
+
+        /**
+         * Scales the data boundary condition by a scalar factor. Can be
+         * used to transform the unit of the data.
+         */
+        const double scale_factor;
+
+        /**
+         * Computes the table indices of each entry in the input data file.
+         * The index depends on dim, grid_dim and the number of components.
+         */
+        TableIndices<dim>
+        compute_table_indices(const unsigned int i) const;
+
+    };
+
+    /**
+     * AsciDataBase is a generic plugin used for declaring and
+     * reading the parameters from the parameter file.
+     */
+    template <int dim>
+    class AsciiDataBase : public SimulatorAccess<dim>
```

Comment:
Out of curiosity, why does this class need to be derived from `SimulatorAccess`? It seems to me that it doesn't, and that if it does that might lead to later problems if (incorrectly) a derived class would derive itself from `SimulatorAccess` again...